Cabal/Distribution/Types/VersionRange.hs

```diff
@@ -440,17 +441,17 @@ instance Parsec VersionRange where
         verOrWild :: CabalParsing m => m (Bool, Version)
         verOrWild = do
             x <- P.integral
-            verLoop (x :)
+            verLoop (DList.singleton x)
 
         -- trailing: wildcard (.y.*) or normal version (optional tags) (.y.z-tag)
-        verLoop :: CabalParsing m => ([Int] -> [Int]) -> m (Bool, Version)
-        verLoop acc = verLoop' acc <|> (tags >> pure (False, mkVersion (acc [])))
+        verLoop :: CabalParsing m => DList.DList Int -> m (Bool, Version)
+        verLoop acc = verLoop' acc <|> (tags >> pure (False, mkVersion (DList.toList acc)))
 
-        verLoop' :: CabalParsing m => ([Int] -> [Int]) -> m (Bool, Version)
+        verLoop' :: CabalParsing m => DList.DList Int -> m (Bool, Version)
         verLoop' acc = do
             _ <- P.char '.'
-            ((\x -> verLoop (acc . (x :))) =<< P.integral)
-                <|> (\_ -> (True, mkVersion (acc []))) <$> P.char '*'
+            ((\x -> verLoop (DList.snoc acc x)) =<< P.integral)
```

Comment:
and here you can drop the lambda on the first line and use `const` on the second line.

Reply:
done, used `<$`